fees/ondo.ts

```diff
@@ -0,0 +1,218 @@
+ * 
+ * No management fees on Ondo for now:
+ * https://docs.ondo.finance/qualified-access-products/ousg/fees-and-taxes#what-fees-does-ousg-charge
+ * https://docs.ondo.finance/general-access-products/usdy/faq/economics-and-fees#what-fees-does-usdy-charge
+ */
+
+const methodology = {
+  Fees: 'Total yields were collected by investment assets.',
+  SupplySideRevenue: 'Total yields were distributed to investors.',
+}
+
+const OndoContracts: any = {
+  [CHAIN.ETHEREUM]: {
+    OUSG: '0x1B19C19393e2d034D8Ff31ff34c81252FcBbee92',
+    OUSGOracle: '0x0502c5ae08E7CD64fe1AEDA7D6e229413eCC6abe',
+    USDY: '0x96F6eF951840721AdBF46Ac996b59E0235CB985C',
+    USDYc: '0xe86845788d6e3E5C2393ADe1a051AE617D974C09',
+    USDYOracle: '0xA0219AA5B31e65Bc920B5b6DFb8EdF0988121De0',
+  },
+  [CHAIN.SOLANA]: {
+    OUSG: 'i7u4r16TcsJTgq1kAG8opmVZyVnAKBwLKu6ZPMwzxNc',
+    USDY: 'A1KLoBrKBde8Ty9qtNQUtq3C2ortoC3u7twggz7sEto6',
+  },
+  [CHAIN.POLYGON]: {
+    OUSG: '0xbA11C5effA33c4D6F8f593CFA394241CfE925811',
+  },
+  [CHAIN.MANTLE]: {
+    USDY: '0x5bE26527e817998A7206475496fDE1E68957c5A6',
+  },
+  [CHAIN.APTOS]: {
+    USDY: '0xcfea864b32833f157f042618bd845145256b1bf4c0da34a7013b76e42daa53cc',
+  },
+  [CHAIN.ARBITRUM]: {
+    USDY: '0x35e050d3C0eC2d29D269a8EcEa763a183bDF9A9D',
+  },
+  [CHAIN.SUI]: {
+    // USDY_TREASURY_CAP_OBJECT_ID
+    USDY: '0x9dca9f57a78fa7f132f95a0cf5c4d1b796836145ead7337da6b94012db62267a',
+  },
+  [CHAIN.NOBLE]: {
+    USDY: 'ausdy',
+  },
+}
+
+const OndoAbis = {
+  totalSupply: 'uint256:totalSupply',
+  getPriceData: 'function getPriceData() view returns (uint256 price, uint256 timestamp)',
+}
+
+// use the same rates fetched on Ethereum for all chains
+async function getPrices(timestamp: number): Promise<{
+  OUSG: number;
+  USDY: number;
+}> {
+  const blockNumber = await getBlockNumber(CHAIN.ETHEREUM, timestamp)
+
+  const [ousgPriceData, usdyPriceData] = await sdk.api2.abi.multiCall({
+    chain: CHAIN.ETHEREUM,
+    abi: OndoAbis.getPriceData,
+    calls: [
+      OndoContracts[CHAIN.ETHEREUM].OUSGOracle,
+      OndoContracts[CHAIN.ETHEREUM].USDYOracle,
+    ],
+    block: blockNumber,
+  })
+
+  return {
+    OUSG: Number(ousgPriceData.price) / 1e18,
+    USDY: Number(usdyPriceData.price) / 1e18,
+  }
+}
+
+async function getSupply(useChainApi: sdk.ChainApi): Promise<{
+  OUSG: number;
+  USDY: number;
+}> {
+  if (useChainApi.chain === CHAIN.SOLANA) {
+    const [supplyOUSG, supplyUSDY] = await Promise.all([
+      solana.getTokenSupply(OndoContracts[CHAIN.SOLANA].OUSG),
+      solana.getTokenSupply(OndoContracts[CHAIN.SOLANA].USDY),
+    ])
+
+    return {
+      OUSG: Number(supplyOUSG),
+      USDY: Number(supplyUSDY),
+    }
+  } else if (useChainApi.chain === CHAIN.APTOS) {
+    const { data: { data: { decimals, supply } } } = await axios.get(`${APTOS_PRC}/v1/accounts/${OndoContracts[CHAIN.APTOS].USDY}/resource/0x1::coin::CoinInfo<${OndoContracts[CHAIN.APTOS].USDY}::usdy::USDY>`)
```

Comment:
can you move this to the helper? this way, we dont repeat the code for future adapters, PR is otherwise good